### ISSUE TITLE
Render tourism=artwork ways

### ISSUE THREAD

**Issue by Tomasz-W** [gravitystorm/openstreetmap-carto#2711]
There is almost 4k of artworks drew as lines:
https://taginfo.openstreetmap.org/tags/tourism=artwork#overview

Those are mostly graffities and murals painted on the walls. Some examples below:
https://www.openstreetmap.org/way/464155454
https://www.openstreetmap.org/way/504508965
https://www.openstreetmap.org/way/370396569

If we render artworks drew as nodes and areas, we should render tourism=artwork ways too, I think. 
For me, the best option would be to render an artwork icon in the middle of these lines.

**Comment by imagico** [comment]
For info: of the ~4k ways with `tourism=artwork` more than 3k are polygons which are already rendered with an icon.

The question to ask is probably more general: in what cases should we also allow linear ways in amenity-points?

**Comment by Tomasz-W** [comment]
@imagico My lists of: 

linear amenities which should be rendered as icons:
- amenity=clock
- amenity=recycling

linear amenities, which should be rendered as icons or icons with lines:
- amenity=bicycle_parking (grey line with current icon in the middle)
- amenity=bicycle_rental (grey line with current icon in the middle)

linear amenities, which should be rendered as lines
- amenity=bench (brown line, https://github.com/gravitystorm/openstreetmap-carto/issues/1890)

**Comment by imagico** [comment]
None of these has documentation on the wiki how to map as linear ways, for most of them the documentation says not to use on linear ways.

Rendering as lines is usually out of the questions because of possible confusion with other line features (highways, railways, barriers etc.).

**Comment by Tomasz-W** [comment]
> for most of them the documentation says not to use on linear ways.

Due to OSM Wiki, it's true just for [clocks ](http://wiki.openstreetmap.org/wiki/Tag:amenity%3Dclock ), [bicycle rentals](http://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbicycle_rental) and [recycling](http://wiki.openstreetmap.org/wiki/Tag:amenity%3Drecycling), so it's not "most of them". All the rest [artworks](http://wiki.openstreetmap.org/wiki/Tag:tourism%3Dartwork), [bicycle parkings](http://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbicycle_parking), [benches ](http://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbench) and [pitches ](http://wiki.openstreetmap.org/wiki/Tag:leisure%3Dpitch)could be mapped as linear ways, so we shoud discuss them.

**Comment by HolgerJeromin** [comment]
If a polygon of an amenity has (also) an icon I propose to render only the icon on lines. Least possible confusion, most gain of shown data.

**Comment by Tomasz-W** [comment]
> The question to ask is probably more general: in what cases should we also allow linear ways in amenity-points?

Due to @imagico comment, I propose to discuss icon-rendering of these elements (mapped as linear ways):

- tourism=artwork
- amenity=bicycle_parking
- amenity=bench
- leisure=pitch

**Comment by HolgerJeromin** [comment]
Ref for bench #1890
Ref for attraction / barrier #1824

**Comment by Tomasz-W** [comment]
Related #844

**Comment by kocio-pl** [comment]
I guess at least tourism=artwork could be added, because wiki treats such nodes, ways and areas the same. I haven't check other tags.

**Comment by Tomasz-W** [comment]
* amenity=bicycle_parking/ rental are sometimes mapped as linear ways, because they are often just a line of standing bikes, e.g.:
![poznanski_rower_miejski](https://user-images.githubusercontent.com/25656654/34839286-609b857e-f701-11e7-8468-626b1d66293a.jpg)
* amenity=bench - there are examples of long benches which is easier to map as lines than nodes
![skwer6](https://user-images.githubusercontent.com/25656654/34839618-8b6b8e06-f702-11e7-9a90-b9f200cf0e17.jpg)
![madrid long bench](https://user-images.githubusercontent.com/25656654/34839646-a70ee68a-f702-11e7-92ea-3ddaaefcf5ea.jpg)

**Comment by matkoniecz** [comment]
> For info: of the ~4k ways with tourism=artwork more than 3k are polygons which are already rendered with an icon.

Do you know how to get this kind of statistics? I see nothing on taginfo and hopefully somebody already shared tool to do that.

For tourism=artwork tagging on lines makes sense and rendering them makes sense. For others I have some doubts but for start I want to check how these are tagged.

**Comment by joshinils** [comment]
according to taginfo, currently these many ways are not shown in any way:
[54 816 amenity=bicycle_parking](https://taginfo.openstreetmap.org/tags/?key=amenity&value=bicycle_parking)
[19 556 amenity=bench](https://taginfo.openstreetmap.org/tags/?key=amenity&value=bench)
[9 427 tourism=artwork](https://taginfo.openstreetmap.org/tags/tourism=artwork#overview)

these are potential tagging mistakes as ways, taginfo does not differentiate areas from lines:
[21 933 amenity=recycling](https://taginfo.openstreetmap.org/tags/amenity=recycling)
[1 540 amenity=bicycle_rental](https://taginfo.openstreetmap.org/tags/amenity=bicycle_rental)
[351 amenity=clock](https://taginfo.openstreetmap.org/tags/amenity=clock)

as for leisure=pitch:
[50 629 leisure=pitch nodes](https://taginfo.openstreetmap.org/tags/?key=leisure&value=pitch) are not rendered.
[leisure=pitch](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dpitch) may not be a way (but an area), but a node is not rendered, [example](https://www.openstreetmap.org/node/5550558012).

**Comment by imagico** [comment]
Keep in mind taginfo lists the number of ways, not the number of linear ways.  To find out the number of linear ways with a certain tag you need to interpret the geometries and the presence of the area=yes/no tag.

**Comment by matkoniecz** [comment]
In my opinion linear ways with amenity=bicycle_parking should remain not rendered, mapping them as area is always better and in case where mapping them as way could be done then mapping them as area was also feasible (except rare exceptions)

**Comment by AndrewKvalheim** [comment]
Example of a [sculpture](https://seattlearts.emuseum.com/objects/1122/flipbooks) that I’ve [mapped](https://www.openstreetmap.org/way/910946528) as a linear way:

[![FlipBooks by Jennifer Dixon](https://user-images.githubusercontent.com/1844746/170795602-fa33b135-f533-4f9f-8758-5098f8f46356.jpg)](https://seattlearts.emuseum.com/objects/1122/flipbooks)

**Comment by 1ec5** [comment]
I’ve [updated](https://wiki.openstreetmap.org/wiki/Special:Diff/2362267) the documentation for `artwork_type=mural` to explicitly allow murals to be mapped as unclosed ways for consistency with several related tags. Currently, unclosed ways [outnumber](https://overpass-turbo.eu/s/1kzb) areas by more than 3 to 1.

**Comment by imagico** [comment]
Yes, `tourism=artwork` is probably relatively speaking the feature most widely mapped with linear ways that we render with a point symbol on nodes and polygons and that would therefore be good to extend to allow linear ways in amenity-points.

Note since many of these will be two node ways good design will call for using `ST_LineInterpolatePoint(way, 0.5)` rather than `ST_PointOnSurface(way)`.

**Comment by richlv** [comment]
Given the general consensus on tourism=artwork, would it make sense to split it out from this issue?
Mapped a couple of murals as ways, considering to revert back to nodes to tag for the renderer ;)

**Comment by jidanni** [comment]
I added a note to https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbench#Rendering .

**Comment by okainov** [comment]
Any updates on this one? What is blocking to improve the render style here?

**Comment by imagico** [comment]
For `tourism=artwork` we have agreement that this would be desirable to render on linear ways.  For some of the other suggestions there is no consensus so far.

The main obstacle is someone would need to go ahead implementing it - and that is difficult in light of #3880.  There are essentially three options:

* actually solving #3880 with support for linear geometries next to nodes and polygons.
* integrating lines (currently separate in the `amenity-line` layer - essentially just ford and slipway, as well as other stuff that does not belong there anyway - see #4104, #4726) into the `amenity-points` layer - that does not solve #3880, but is a cleaner approach - would be wasted work though if #3880 is to be addressed at some point.
* lazily adding it to `amenity-line` - would just pile onto existing problems so not likely seen very favorably by anyone.

Bottom line: what is blocking this is our over-extension in technical debt, which makes it hard find the legroom for meaningful design changes.
